File: 2025/05/main_test.cpp
#include <print>
#include <string>
#include <vector>

#include "lib.cpp"

void assert(int expected, int actual) {
    if (expected == actual)
        std::println("Assertion Passed.");

    else
        std::println("Assertion Failed; expected: {}, actual: {} ", expected, actual);
}

int main() {
    std::vector<std::string> data = {
        "3-5", "10-14", "16-20", "12-18", "", "1", "5", "8", "11", "17", "32",
    };

    auto first_expected = 3;
    auto first_result = first(data);
    assert(first_expected, first_result);
}

File: 2025/06/main_test.ts
import { first, second } from "./lib.ts";

function assert(expected: number, actual: number) {
    if (expected == actual) console.info("Assertion Passed.");
    else console.error(`Assertion Failed; expected: ${expected}, actual: ${actual}`);
}

function test_main() {
    const data: string[] = ["123 328  51 64 ", " 45 64  387 23 ", "  6 98  215 314", "*   +   *   +  "];

    const firstExpected = 4277556;
    const firstActual = first(data.slice());
    assert(firstExpected, firstActual);

    const secondExpected = 3263827;
    const secondActual = second(data.slice());
    assert(secondExpected, secondActual);
}

test_main();
